key/aziot-keys/src/lib.rs

```diff
@@ -159,69 +159,187 @@ pub struct AZIOT_KEYS_FUNCTION_LIST_2_0_0_0 {
     ///
     /// If an implementation does not recognize a particular component as an algorithm, or is unable to use the algorithm to generate a key pair,
     /// it should ignore that component and try the next one. If no components are left, the implementation should return an error.
-    /// It is allowed for the implementation to unable to generate a key pair even if the wildcard algorithm is specified.
-    ///
-    /// If `preferred_algorithms` is NULL, it should be interpreted the same as if it was `"*"`.
+    /// The implementation is allowed to be unable to generate a key pair regardless of which algorithms are specified; this is true even if
+    /// the wildcard algorithm is specified.
     ///
-    /// The public key is written to `ppublic_key` and the private key to `pprivate_key`.
-    /// For keys generated by openssl in memory, the private and public components of a key live in the same `EVP_PKEY` value.
-    /// However keys loaded from engines do differentiate between the two, so separate `EVP_PKEY` values are required.
-    /// Even if the implementation generates keys in memory using openssl, it must copy the public parameters out of the key into a new `EVP_PKEY`
-    /// and set `ppublic_parameters` to that.
+    /// If `preferred_algorithms` is `NULL`, it should be interpreted the same as if it was `"*"`.
```

Comment:
Is "should" being used in the [RFC2119](https://tools.ietf.org/html/rfc2119) sense? i.e: is it fine for an implementation to not do that? If not, "MUST" would be more appropriate.

Reply:
Yes, it's a hard requirement. Changed the `should`s to `must`s.